### PR TITLE
I/O container improvements

### PR DESCRIPTION
**Pull request by kemp-google**
This PR adds exponential backoff and another workaround for gsutil bugs.

### CODE REVIEW

File: io/retry/retry.go [hunk 1]
```diff
@@ -35,32 +34,40 @@ func main() {
...
 		command.Stdout = os.Stdout
 
-		if err := command.Run(); err != nil {
-			if !isRetriable(tool, err) {
-				if err, ok := err.(*exec.ExitError); ok {
-					os.Exit(err.Sys().(syscall.WaitStatus).ExitStatus())
+		switch err := command.Run().(type) {
+		case *exec.ExitError:
+			status := err.Sys().(syscall.WaitStatus).ExitStatus()
+			if attempts < 3 {
```
Comment (tjennison-google) [inline io/retry/retry.go]:
```
if attempt >= 2 {
  os.Exit(status)
}
```

And remove continue below.

File: io/retry/retry.go [hunk 2]
```diff
@@ -35,32 +34,40 @@ func main() {
 	}
 
 	tool := os.Args[1]
+	arguments := os.Args[2:]
 
-	for attempts := 3; attempts > 0; attempts-- {
-		command := exec.Command(tool, os.Args[2:]...)
+	var attempts int
+	for {
```
Comment (tjennison-google) [inline io/retry/retry.go]:
```
for attempt := 0; ; attempt++ {
```

And remove attempts++ below.

File: io/retry/retry.go (hunk 1 above)
Comment (kemp-google) [reply io/retry/retry.go]: Done

File: io/retry/retry.go (hunk 2 above)
Comment (kemp-google) [reply io/retry/retry.go]: Done

**Pull request merged**